packages/rum/src/domain/segmentCollection/segmentCollection.ts

```diff
@@ -125,12 +127,21 @@ export function doStartSegmentCollection(
       creationReason,
       initialRecord,
       (compressedSegmentBytesCount) => {
-        if (!segment.isFlushed && compressedSegmentBytesCount > SEGMENT_BYTES_LIMIT) {
+        if (!segment.flushReason && compressedSegmentBytesCount > SEGMENT_BYTES_LIMIT) {
           flushSegment('segment_bytes_limit')
         }
       },
       (data, rawSegmentBytesCount) => {
-        send(data, segment.metadata, rawSegmentBytesCount)
+        const payload = buildReplayPayload(data, segment.metadata, rawSegmentBytesCount)
+        if (
+          !isExperimentalFeatureEnabled('retry_replay') ||
+          segment.flushReason === 'visibility_hidden' ||
+          segment.flushReason === 'before_unload'
+        ) {
+          httpRequest.sendOnExit(payload)
+        } else {
+          httpRequest.send(payload)
+        }
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 praise:** ‏Nice!